### ISSUE TITLE
Reintroduce some options removed with v7

### ISSUE THREAD

**Issue by lwouis** [lwouis/alt-tab-macos#3882]
Hi everyone,

When we introduced v7, the big goal was to #351. I think we achieved that. Thanks to @Allsochen's hard work, we were able to move from a page that 1% of users understood how to use, to something that looks like it was designed by Apple themselves.

Before v7 | After v7
-|-
![image](https://github.com/user-attachments/assets/b0c68336-92cd-4ead-b1da-4dcd763836de) | ![image](https://github.com/user-attachments/assets/e352964e-d413-4f9b-8559-5a6b2d842503)

We've added many new features as well. Added visual illustrations of advanced features, etc. It's a huge win overall for most users.

Along the way, we've had to remove some advanced customization. We removed it because it was too complex and a detriment for the vast majority of users. We also removed it in some cases because it was not a good idea to keep it and have to maintain it on our side, as well as to support users opening tickets with very exotic setups. Context for why manpower on our side is an important consideration: #1179 

After v7 release, many people have sent feedback that they noticed a feature was gone, and they wanted it back. This was expected. We're now looking at how many people ask for certain features to be back, and are considering reintroducing a select few. This ticket tracks this, as well as being a notice we can redirect people to.

Current candidates we are considering:

* Mouse follow focus => Added back ✅ 
* Being able to adjust the switcher width => auto-width was added. It decides the width based on ergonomic field-of-view. It should help with most cases with large screens. Not sure if we want to add more than this.
* Adding an XS and XL Appearance Size => sounds like we should add it.
* Providing a Style with App Icons and Titles, with all Titles visible at once => Maybe we could update the Thumbnails Style to be more ergonomic when the Screen Recording permission is not given? It would be a bit hidden though. Maybe it could be an option of the Thumbnails Style, in addition? ("no thumbnails"). Not sure. 
* Improve min-width in Titles Style => this one we should definitely do
* Let users use different Styles with different shortcuts => this one seems to complex to happen. Both at the UX level in the preferences (how to set this?), and at other levels. I don't think it will happen
* Number of shortcuts?

Thank you 🙇

**Comment by ezhang7423** [comment]
Thank you so all much for the hard work. For myself personally, "Mouse follow focus" would be absolutely huge

**Comment by TechTronicLLC** [comment]
First, thanks for the effort in making such a great app free for all to enjoy. I just installed AltTab specifically and solely for the "Mouse follows focus" feature after Googling to find a better solution for managing apps across multiple displays. I was a bit disappointed that the feature was removed and would greatly appreciate if it returned.

**Comment by StephenHaney** [comment]
The new settings are much easier to use!

Some settings I used previously that I miss:
- Max width or max-windows-per-row for the picker (on my widescreen monitor it's ... really wide)
- When using cmd + ` I'd like the focus to happen immediately, not on release

Less important:
- Prevent shift key from immediately going backward, require shift + tab to go backwards like native picker

**Comment by lwouis** [comment]
> Max width or max-windows-per-row for the picker (on my widescreen monitor it's ... really wide)

Could you please share a screenshot of how it looks on your setup?

Thank you

**Comment by StephenHaney** [comment]
> > Max width or max-windows-per-row for the picker (on my widescreen monitor it's ... really wide)
> 
> Could you please share a screenshot of how it looks on your setup?
> 
> Thank you

Sure, here is an example. I usually have three windows across this screen. I'd probably expect or want around 10 icons of width.
![image](https://github.com/user-attachments/assets/b3e3ec51-ba87-43a5-9066-2c36a89f1b30)

**Comment by gpetretto** [comment]
Hi, first of all I would like to thank you for the great work on writing and maintaining this app. 
As you are collecting opinions, I would also really appreciate having an option to see the app Icons and titles, with all titles visible at once. In a situation like the one in the screenshot above it is much more difficult to identify which Chrome window to target, compated to how it was in v6.

**Comment by hicaicai** [comment]
I would like to be able to set the maximum number of apps to display，Because if there are too many apps, it is difficult to choose. It is much easier than setting custom width and length.

**Comment by Tinyu-Zhao** [comment]
Hope to support "Being able to adjust the switcher width"

**Comment by tirumala007** [comment]
this is great. can we get Windows Feature to open apps like Windows+1 opens 1st app,Windows+2 opens the 2nd app.
There's an app called "Snap" on App Store. it works but not good. can you add that feature to your app?

**Comment by slieschke** [comment]
> Max width or max-windows-per-row for the picker (on my widescreen monitor it's ... really wide)

Indeed. I use the Titles appearance and also find the switcher width excessive on my widescreen monitor (see screenshot). It would be great to be able to reduce the width in some fashion so the title text displays more centrally on the screen.

![CleanShot 2024-12-10 at 10 43 21](https://github.com/user-attachments/assets/4259fcf7-e707-47a1-8a66-15ef85b42d9e)

**Comment by be9em0t** [comment]
> * Prevent shift key from immediately going backward, require shift + tab to go backwards like native picker

@StephenHaney, this functionality is availabe if you define Shift-Tab as backwards key (Shortcuts When Active -> Select previous window)

**Comment by be9em0t** [comment]
@lwouis I join the chorus of thanks for this great program!

I would like to see the icon scaling feature made available again.

Currently we can only scale windows, but the application icons remain very small and hard to recognise.
With previous versions of AltTab I would always scale up the icon to a size similar to the default Macos window switcher.

If you think this feature is confusing to users you can probably hide it somewhere deeper in the UI 😸

**Comment by vladunev-tomtom** [comment]
> I would like to see the icon scaling feature made available again.
> 
> Currently we can only scale windows, but the application icons remain very small and hard to recognise. With previous versions of AltTab I would always scale up the icon to a size similar to the default Macos window switcher.

To support my ask, here is a snapshot of how I used to run AltTab 6. I miss this very much...
![AltTab icon size](https://github.com/user-attachments/assets/a3fe7560-49c9-410b-9ffa-f67d473d69d8)

**Comment by esserrge** [comment]
+1 for adjusting the switcher width.

4K 28":
![Screenshot 2024-12-14 at 13 37 47](https://github.com/user-attachments/assets/2ca63d81-5cb8-4263-ae28-e04a4bac7ef6)

**Comment by kaitpw** [comment]
I very much miss changing the app icon's size. I used to have my icons and the thumbnail at similar sizes like @vladunev-tomtom. A sm/md/lg picker for that would be sweet. 

Also- I don't super mind the AltTab window's width but I did used to have mine narrower which I preferred. 

Great work ya'll. Despite the changes this is a go to for me.

**Comment by xzbdmw** [comment]
My two cents, the Titles appearance are too tall if there are too many open windows, and in most cases the highest few items will be selected, so aside of width customization, the option to say how many items are showing at one page is much appreciated.

**Comment by qns7** [comment]
Thank you VERY much for adding the "mouse/cursor_follows_focus" back!

**Comment by pgsill** [comment]
Under the hood are the old options still present, just without UI?
An "Advanced" panel would be a great way to let us nitpickers adjust the app to our use!

**Comment by fbeccaceci** [comment]
Plans on adding back a max-width?

**Comment by lwouis** [comment]
@fbeccaceci This is discussed above. Please read the thread.

**Comment by felixier** [comment]
I have issues with using only 3 shortcuts.
But I have an interesting reason: I have ISO keyboard on my Mac, while I have an  external ANSI magic keyboard keyboard and again an older ISO HP keyboard at home. I use combination option + key next to 1 below Esc. But there is a different key on ISO vs ANSI (only on Mac, not on PC, btw.). So basically currently I keep manually changing keyboards while working on different keyboard.
It is easier than using different keys on different keyboards (namingly key right from left shift on ISO keyboard).
With more shortcuts, I had some duplicity, but it worked well on different keyboard layouts.

**Comment by lwouis** [comment]
People are exposing their use-cases here. Discussion will help decide next steps. Please be patient. we need to review things. Also please consider the pinned issue #1179 in the picture.

**Comment by ADTC** [comment]
+1 on max-width for me also. Or somehow make it more sensible than just the "width based on ergonomic field-of-view".

And, "Titles" mode should have option to have 2 or 3 columns. Preferably, with minimum number of items filled in one column, before opening a new column.

And explanations (info icons ℹ️) for what these are:
* Visibility
* Cursor follows focus

I couldn't understand them just by reading the label.

**Comment by etchevarneM** [comment]
Yeah, it is too wide, specially in Title mode

<!-- Uploading "too wide.png"... - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ->

**Comment by etchevarneM** [comment]
Yeah, it is too wide, specially in Title mode

<!-- Uploading "too wide.png"... - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ->

**Comment by jchayat** [comment]
Please consider increasing the number of shortcuts. Better yet, allow a variable number of shortcuts (using a list UI vs a tab UI).

I used to have 4 shortcuts configured for example: alt-tab for cycling through all windows of all apps, alt-` for cycling through all windows of the current app, as well as ctrl-alt-tab and ctrl-alt-` correspondingly for limiting to the current space.  

Now the app has been limited to 3 shortcut, so I had to give up one of those use cases.

**Comment by snk-actian** [comment]
> Please consider increasing the number of shortcuts. Better yet, allow a variable number of shortcuts (using a list UI vs a tab UI).
> 
> I used to have 4 shortcuts configured for example: alt-tab for cycling through all windows of all apps, alt-` for cycling through all windows of the current app, as well as ctrl-alt-tab and ctrl-alt-` correspondingly for limiting to the current space.
> 
> Now the app has been limited to 3 shortcut, so I had to give up one of those use cases.

➕ 

Same situation here, at least getting back the fourth shortcut would be great.

**Comment by RiccardoAgatea-Moku** [comment]
I really miss the option to change the font size of window titles independently from the preview size. I'm nearsighted, so I struggle with small text. I use the "Thumbnails" style, and I need to keep it large for the titles to be at a comfortable size, but this way I only see a few windows at a time.

Like some others said above, I'd like some way to access the old options, even if it's a bit buried in the UI.

**Comment by phazei** [comment]
Despite all the complaints, this app is really incredible and all the work is incredibly appreciated.  I'm a web developer, and I used to use multiple large monitors, but I've transitioned and now primarily program on a 13" MBP with a LOT of desktops, without this app, it simply would not be remotely possible for me to handle this setup.  So just wanted to give a big thank you!

I'm glad we're having a big discussion and seeing lots of user feedback and use cases.  I don't know what level of technical aptitude most people using the app have, I realize mine is generally on the higher side, I'm the type to run custom phone roms and have config apps with all the kernel settings and will sit down and look through every option to know what does what, but I realize a lot of people don't have that level of patience, and it's quite the antithesis to Apple's design paradigme, but advanced buttons are golden ;)

**Comment by lwouis** [comment]
@antran22 thank you
You can have a grid-like experience today in Thumbnails Style. Do you mean you would likr a grid in Titles Style?

**Comment by pauljmyers** [comment]
I will add another voice behind allowing more shortcuts. I was extremely disappointed to go from five on a previous version to now only three on one of my machines. Personally, I do not seeing myself ever using the "Gesture" shortcut -- I'd much rather use the Mac default gestures with swipes, etc.  -- and would much rather have at least a fourth customizable keyboard shortcut instead.

(Now even with that said, this app is incredible, especially given that it is available for free, so thank you!! It is easily my most-used app next to Rectangle, and it made the switch from Windows a few years back supremely painless.)

**Comment by azhongzl** [comment]
Thank @lwouis for such a great app.

It would be fantastic if we can adjust the window size (like what we supported before). So far the window (even with "Large" size) is too small for me.

<img width="561" alt="Image" src="https://github.com/user-attachments/assets/9066fdd0-2920-4ec2-a26c-f36254696c98" />

**Comment by billy-bg** [comment]
I will also add a show of support for increasing the number of (non-gesture) shortcuts back to 5. This number of shortcuts is especially useful with multi-monitor setups, which is where i believe this app really shines.

I don't believe this should make the settings harder to understand given the fact that it is not adding any _new_ information that the user has to understand. I am also confident that given your past UI design experience, this can be added with minimal additional clutter as a side effect (can either reduce white space slightly and make settings window slightly wider or add a horizontal scroll bar).

I would also argue that more users would prefer keyboard over gesture shortcuts given the following:
  1. The default macOS settings for gestures are more customisable (i.e. "good enough" for most users) than the default macOS keyboard window switching settings (essentially non-existent).
  2. I would imagine a larger percentage of Alt-Tab users have multi-monitor setups in which they use a keyboard/mouse over a trackpad.

I've always found your UI design better than most apps even before the most recent changes (i.e. compared to rectangle), and have always found it very easy to understand while still striking a good balance between functionality and simplicity. I would also doubt that only 1% of users understood the previous settings page (just a hunch, but people who are happy with the UI are much less likely to message you compared to people who are unhappy with it, so there may be a reporting bias here).

Regardless, thank you for your continued work and attention on this excellent application 💚

**Comment by GianmarcoFolchi** [comment]
Just wanted to reply to say that I am currently still using Version 6 due to V7.xxx removing the ability to have horizontal switcher without thumbnails

**Comment by antran22** [comment]
> [@antran22](https://github.com/antran22) thank you You can have a grid-like experience today in Thumbnails Style. Do you mean you would likr a grid in Titles Style?

Hi ya, I see that Thumbnails style is indeed grid like. However, most of the time, with only 6-7 windows per workspace, it's still linear. With a customized grid layout option, I'll be able to navigate these windows with 3-4 keystroke at most (using hjkl or arrow). Without grid arrangement, to get to the final windows I'll need to press 7 keystrokes. So this might give you some idea on how certain people might view your app in terms of ergonomics.

**Comment by Elijas** [comment]
@lwouis 
More shortcuts is a **must-have** for multi-screen setups. It's also very easy to keep the UI fully simple and apple-like

![Image](https://github.com/user-attachments/assets/dfed733f-9604-4bc9-a48e-0a4c716699b4)

**Comment by Elijas** [comment]
@lwouis 
More shortcuts is a **must-have** for multi-screen setups. It's easily among the top 2 of the requested features that are requested to be added back to v7 in this thread.

It's also very easy to keep the UI fully simple and apple-like:

![Image](https://github.com/user-attachments/assets/dfed733f-9604-4bc9-a48e-0a4c716699b4)

@nikkundra @felixier @jchayat @snk-actian @pauljmyers @smzelek Feedback welcome

**Comment by jchayat** [comment]
@Elijas I think this looks great. 
@lwouis I think these’s enough traction for this feature request. WDYT?

**Comment by lwouis** [comment]
Hi,

The mockup above is pretty cool. Here's some feedback:

* We can't merge the hold and press sections. The way alt+tab works, you will hold some keys, and release other keys. We need to keep the 2 groups separated. People use the flexibility
* The mockups are missing the current Gesture tab (trackpad)
* The mockups are missing the current `Additional controls...` and `Shortcuts when active` buttons
* I think showing the keys next to each shortcut is a nice and useful addition
* I don't think we would add a UI where the user adds shortcuts. I think hardcoding is the way to go. It's simpler. The question in my mind is more: how many tabs is the right amount. I think today 3 (keyboard) + 1 (trackpad) is good for 99% users probably. There are not that many people in this thread asking for more tabs, actually. If we would add more, I think we would just have 4 or 5. Beyond that, it's really too niche, and it brings complexity with the tab management (e.g. the left-side on the mockups would need to be a scrollable list).

![Image](https://github.com/user-attachments/assets/8ee11b6d-f8e8-45bb-9df0-bdee7744f4d6)

Thank you

**Comment by jchayat** [comment]
@lwouis Most users are silent and do not bother commenting, particularly if they see someone else asking for the thing they wast (I know I probably wouldn’t). So, really having a few people expressing their preference should be viewed as representing a larger population. I would also point out that I have not seen any comment objecting to an increase. There really is no way to know what 99% of the users want from a discussion like this. One can only hope to get a stochastic “bump” in the right direction.

**Comment by phazei** [comment]
I keep checking this thread hoping a 4th shortcut is added.  I've seen a lot of tickets and comments requesting it.  I need the 4th and have been stuck on an old version until that happens.

**Comment by Elijas** [comment]
Hi @lwouis,

Thanks for engaging and for your work on AltTab – it's a fantastic app.

Regarding shortcuts: I understand your take on the numbers, but for many of us, this isn't a niche request. The need for at least 4 shortcuts (e.g., all/current app x all/current space/screen) is a core reason AltTab is so powerful, and it's why many, myself included, have struggled with the reduction or reverted to older versions. This was a key workflow.

A hardcoded increase to 4, or ideally 5, keyboard shortcuts would be a massive improvement and wouldn't necessarily bring back the complexity of v6's settings. It would solve a major pain point for a significant portion of your dedicated users.

Appreciate you considering this.

---

P.S.
Regarding your feedback on the mockup (missing hold/press, gestures, other buttons): You're absolutely right, those are all essential. My mockup was intentionally simplified to focus *only* on how more shortcut *tabs* could be presented, not to imply the other crucial controls aren't needed.

P.P.S.
Alternatively, have you ever considered an optional "Advanced Settings" panel or a "Power User Mode"? This could be a way to bring back more granular controls or a higher shortcut count for those who need them, without cluttering the default UI for the majority. It might be a good middle ground and could help keep power users on the main release rather than needing to maintain forks for specific features. For the shortcut count, simply increasing the hardcoded number of available tabs within an "Advanced Settings" panel might be a manageable way to address this, leveraging existing code.

**Comment by SoCuul** [comment]
It would be nice to have a checkbox to enable/disable the shortcuts. As if someone only wants the first shortcut, there's no way to disable the 2nd and 3rd ones without setting a shortcut combination that wouldn't be hit by accident.

**Comment by ADTC** [comment]
@SoCuul if I recall correctly you can clear the shortcut field (I don't remember exactly how) to disable the shortcut. Not as convenient as a checkbox (that would be nice!) but it's possible.

**Comment by KooliFab** [comment]
<img width="551" alt="Image" src="https://github.com/user-attachments/assets/f5c81e5e-1364-47c3-8f6b-6df3d63bccec" />

Is it possible to chose between "Show" and "After release" 🙏

**Comment by vladunev-tomtom** [comment]
Isn't this the same as Appearance tab -> Customize Thumbnails style... -> Preview selected window?

**Comment by lwouis** [comment]
@KooliFab I'm sorry, I don't understand what you mean

**Comment by vladunev-tomtom** [comment]
@KooliFab, I think you want to Show (preview) the selected application window ast you ctrl-tab. 

This option you'll find under Appearance, then click on Customize Thumbnails style... and then select the last option,  Preview selected window.

**Comment by vladunev-tomtom** [comment]
Sorry, now I've lost you :)

<img width="531" alt="Image" src="https://github.com/user-attachments/assets/241cadee-992f-42bb-bdd1-2f39831df434" />

**Comment by KooliFab** [comment]
😁 I just want the preview window to stay open after I release the key, until I select an app.

**Comment by lwouis** [comment]
@KooliFab see screenshot above. This will do it

**Comment by KooliFab** [comment]
My bad, I was testing with the wrong shortcut 😅. The feature is working fine 👍 Thanks

**Comment by SoCuul** [comment]
> Indeed you can click the right box of the shortcut and then click the little cross to empty. This disables the shortcut

@lwouis ah, that's definitely hard to notice without knowing you have to click on the shortcut box first. maybe the x icon could just always show as long as the shortcut box is not empty?

This is a quick little concept of what I think would work well. Took some inspiration from the macOS built-in keyboard shortcut selection panel.

It would also be nice to have the same option to disable the left shortcut modifier (like the right one), as some people may be under the misinterpretation that it means it would activate by just pressing the modifier key.

![Existing Shortcut](https://github.com/user-attachments/assets/14106e5d-4f8e-461c-a6bd-70649982b10b)

![Disabled Shortcut](https://github.com/user-attachments/assets/45e94f56-e09f-4895-a801-5a608012ec99)

**Comment by marijnbent** [comment]
To add to https://github.com/lwouis/alt-tab-macos/issues/4094, I would love to limit how wide the switcher is. I'm using the tiles function, and if I have a browser tab open that has a long title, it sometimes covers most of the width of my screen. I think a sensible default will be the current auto mode, but also give the option to select a fixed percentage (from 20 to 80, with increments of 10?) instead.

Another option is for the user to select between auto, narrow, standard, or wide.

**Comment by bartier** [comment]
Hello everyone, I created for myself a fork and implemented per-shortcut switcher style at https://github.com/lwouis/alt-tab-macos/pull/4534

I am using for myself and all good so far. It makes more sense for my user flow to have alt tab working as macOS and alt + ` to work to show windows.

BTW> not a Swift developer. Just a SRE guy that was bothered about the behavior. Used some LLM to discover quickly the codebase.

**Comment by moritzluedtke** [comment]
Also +1 from my side for limiting the max width of the switcher. And I fully agree with what @marijnbent suggested in his last comment.

I'm using it without big app icons so I just have the list of apps and for me personally it is too wide like this on an ultra wide monitor. The monitor I have at work has a resolution of 5120x1440.

The screenshot is rather basic because it's a company device and I need to limit what can be seen:
![Image](https://github.com/user-attachments/assets/081f16fa-8824-4827-9bbe-07fc3068513a)

(Note, the menu bar from MacOS is cut off at the top)

Thank you for your great work on this app! I really like the new settings window and for me this is the only thing that is missing.

**Comment by frypf** [comment]
+1 from me for the return of Shortcut 4 please - I was using shortcuts 3 and 4 for show all windows / app windows via different gestures in BetterTouchTool. I appreciate that there is now a gesture option, but it's very limited compared with BTT.

(I hadn't even thought to check AltTab as to why this wasn't working, I assumed it was just some idiosyncrasy of my convoluted BTT setup 🤷‍♂️)

**Comment by J-Siu** [comment]
> Also +1 from my side for limiting the max width of the switcher. And I fully agree with what [@marijnbent](https://github.com/marijnbent) suggested in his last comment.
> 
> I'm using it without big app icons so I just have the list of apps and for me personally it is too wide like this on an ultra wide monitor. The monitor I have at work has a resolution of 5120x1440.
> 
> The screenshot is rather basic because it's a company device and I need to limit what can be seen: ![Image](https://github.com/user-attachments/assets/081f16fa-8824-4827-9bbe-07fc3068513a)
> 
> (Note, the menu bar from MacOS is cut off at the top)
> 
> Thank you for your great work on this app! I really like the new settings window and for me this is the only thing that is missing.

+1.

I opened suggestion last year, then got merged into https://github.com/lwouis/alt-tab-macos/issues/3770#issuecomment-2445580299

Really want an auto fit option.

**Comment by lwouis** [comment]
I would like to have Titles style auto-fit as well. If someone can contribute a PR, i would merge it for sure 👍

**Comment by IanAtSalt** [comment]
Can we restore the "show tabs as windows option"? 
It seems like more and more, I'm having to use Chrome based web apps, and its tiresome to alt-tab to Chrome and THEN have to find the proper tab where the app lives. 

As a workaround, I'm trying to create a ton of Chrome PWAs and switch to those, but would love to have in as an option in Alt-Tab.

**Comment by lwouis** [comment]
I worked on Liquid Glass support 🚀 I've reworked the Appearance to remove Visibility.

Could you please try out [this build](https://github.com/user-attachments/files/23344673/AltTab.app.zip)?

I would be very grateful if someone not yet on macOS Tahoe could test this build!

Thank you

**Comment by be9em0t** [comment]
Testing [this build above](https://github.com/user-attachments/files/23344673/AltTab.app.zip).
But I'm on Tahoe.
Also, I dont't realy understand what "I've reworked the Appearance to remove Visibility" means.
One minor thing so far - in Icons mode custom icons get slightly cut off (sorry, can't find a way to screenshot).

**Comment by lwouis** [comment]
You can launch Screenshot.app and set a timer. This way you can screenshot AltTab 👍

**Comment by ADTC** [comment]
@lwouis thank you for this app. I use it throughout my entire day of working, every day. It's my most used third-party app besides Google Chrome on my Mac, and I absolutely cannot live without it! THANK YOU!

**Comment by lwouis** [comment]
Thank you @ADTC! As a long timer user, how do you like the Liquid Glass look in the build i shared above?

**Comment by phazei** [comment]
Any chance you've adjusted the design in some fashion that we can get the 4 shortcuts back?

**Comment by ADTC** [comment]
@lwouis sorry I wish I could give you feedback but I just keep sticking to Sonoma even though Tahoe is already out. I didn't find a reason to upgrade yet. 😆... I am sure it would look great though.

**Comment by Elijas** [comment]
> Any chance you've adjusted the design in some fashion that we can get the 4 shortcuts back?

yes, four shortcuts would be preferable
I use three all the time
- `Cmd+Tab` for switch with consistent order, only for specific monitor (for navigating by mouse hover)
- `Opt+Tab` for switch by last used, only for specific monitor (for navigating only by keyboard only)
- `Cmd+backtick` for switch with consistent order between ALL windows (all monitors) (for navigating by mouse hover)
So three is the bare minimum, I would love to experiment with fourth (`Opt+backtick`)

**Comment by lwouis** [comment]
@ADTC actually it would help a lot if you would test this build on your setup. I can't test old macOS on mine, and i worry about breaking compatibility 🙏

**Comment by lwouis** [comment]
Hi,

Could you please try out [this local build](https://github.com/user-attachments/files/23436971/AltTab.app.zip)? It contains:

* Titles Appearance auto-width, as discussed in https://github.com/lwouis/alt-tab-macos/pull/4908#issuecomment-3507794689
* Tahoe and Liquid Glass new look for the switcher
* Improved responsiveness of showing the switcher

Thank you

**Comment by rosmur** [comment]
hi  @lwouis,

just checking in if allowing text size control (in Titles mode) is still planned? I had raised #4898 a while back, but it was closed in lieu of this issue

**Comment by lwouis** [comment]
There may be "Adding an XS and XL Appearance Size" if enough people ask. For now, no one has asked for months

**Comment by phazei** [comment]
How many people would have to ask to have 4 shortcuts?  Because I've seen quite a lot

**Comment by Elijas** [comment]
> How many people would have to ask to have 4 shortcuts? Because I've seen quite a lot

@phazei I'll try to objectively systemize the data.

# Samples represent a larger user base

> jchayat (May 17) - Most users are silent and do not bother commenting, particularly if they see someone else asking for the thing they want (I know I probably wouldn’t). So, really having a few people expressing their preference should be viewed as representing a larger population. I would also point out that I have not seen any comment objecting to an increase. There really is no way to know what 99% of the users want from a discussion like this. One can only hope to get a stochastic “bump” in the right direction.

# Unique users requesting this 

@nikkundra (Dec 14, 2024) - "Would love to have Shortcuts 4 and 5 back"
@felixier  (Jan 14) - "I have issues with using only 3 shortcuts"
@jchayat (Jan 20) - Asked for increasing shortcuts, had 4 configured before
@snk-actian  (Jan 20) - "at least getting back the fourth shortcut would be great"
@pauljmyers (Jan 28) - "disappointed to go from five on a previous version to now only three"
@billy-bg (Mar 2) - "show of support for increasing the number of (non-gesture) shortcuts back to 5"
@smzelek (Mar 10) - "had to revert to using v7.10.0 so I could use more shortcuts again"
@Elijas (May 17) - "More shortcuts is a must-have for multi-screen setups"
@phazei (May 18) - "need the 4th and have been stuck on an old version"
@frypf (Aug 10) - "+1 from me for the return of Shortcut 4 please"
@phazei (6 hours ago) - "How many people would have to ask to have 4 shortcuts? Because I've seen quite a lot"_

@wtfzambo https://github.com/lwouis/alt-tab-macos/issues/3995 - "I was using all 5 different shortcut options and was quite disappointed to see that in latest major release, 2 of them were gone."
@GJRobert https://github.com/lwouis/alt-tab-macos/issues/1064 - "Please kindly consider add a couple of new [shorcut] setting slots"
@vinodt1 https://github.com/lwouis/alt-tab-macos/issues/4489 - "I think having at least 5 shortcuts is useful in my opinion. Currently only 3 shortcuts are there."
@PrayagS https://github.com/lwouis/alt-tab-macos/issues/1064 - "Can you please let us know your thoughts on this [shortcuts request]? Is this blocked by something?"
@daniel-zeng + @decodism + @eugenesvk https://github.com/lwouis/alt-tab-macos/issues/1064 - "I think having a hardcoded 4 or 6 would definitely be currently sufficient, if it's not too difficult to update. Thank you!"
@anton-kapelyushok https://github.com/lwouis/alt-tab-macos/issues/1933 - "Increase number of available shortcuts"
@shailesh17 https://github.com/lwouis/alt-tab-macos/issues/1988 - "Can we add one or two more shortcuts so we can do different combinations of control settings?"

Some samples from related threads that show that having many shorcuts is fundamental to the UX experience of the app:

@norrbru https://github.com/lwouis/alt-tab-macos/issues/3766 - "make one shortcut that shows plain windows and another that shows the same windows with detailed status indicators."
@rxhfcy https://github.com/lwouis/alt-tab-macos/issues/463 - "My particular obscure preference would be to be able to switch between all windows with cmd+tab, and between windows of the current program with alt+tab...

# Use Cases
- **Different visual configurations per shortcut.** Essential for multi-space, multi-screen and other more many-window setups. For example:
    - @norrbru: Wanted plain view vs detailed status indicators tied to different shortcuts
    - @rxhfcy: Wanted cmd+tab for all windows, alt+tab for current app windows (different modifier keys)
    - @bartier https://github.com/lwouis/alt-tab-macos/pull/4534 - Built a fork implementing per-shortcut "Show in Switcher" to enable ⌘Tab for all apps, ⌘` for current app windows
- **Keyboard layout conflicts.** Workaround for AltTab supporting different languages/layouts. For example:
    - @felixier: Has both ISO and ANSI keyboards; used duplicate shortcuts across different layouts to maintain consistent behavior regardless of which keyboard is active
- **Automation integration (shortcuts-as-a-bridge):** Useful for mapping gestures and special hardware buttons. For example:
    - @vinodt1: Wanted invokable shortcuts for integration with Better Touch Tool and Logi Options+ mouse gestures

**Comment by billy-bg** [comment]
excellent post @Elijas , i am still using the old version of Alt-Tab with 5 shortcuts at this point, so would love for the number of shortcuts to be updated back to 5+ again.

**Comment by TheophileMot** [comment]
For what it's worth, I'm still using the old version (v6.73.0) as well. I'm one of the apparently 1% of users who understood the old design, and I've been following this thread in the hope that one day all the good old options will be restored.

**Comment by Elijas** [comment]
> For what it's worth, I'm still using the old version (v6.73.0) as well. I'm one of the apparently 1% of users who understood the old design, and I've been following this thread in the hope that one day all the good old options will be restored.

What old options you wish were brought back? (High/low importance)

**Comment by felixier** [comment]
I still don't get why it is such a big issue to add one or two more shortcuts.
Will it really make the UI (and UX) unusable for (99% users}?
And as pointed by @Elijas above, there are many users asking to get more shortcuts...
I would even say those who are OK with 2-3 shortcuts may even be OK with using the build-in macOS shortcuts (Cmd+tab).

**Comment by enderunal** [comment]
Hello,

I love the app and have been using it for a long time. Although I value your effort and making it easier for generic user I think the previous versions had better ability to configure your app. See how I use it: 

<img width="581" height="710" alt="Image" src="https://github.com/user-attachments/assets/2a464dcb-9e93-457b-bd27-081d976da352" />

The new version does not give you the flexibility. To be honest I've immediately switched back to latest V6 release and I'll never upgrade until there is a better way to make the UI smaller and/or configurable.

So, I'm hoping that you could provide the smaller and minimalistic aesthetics which were possible with the versions below 7. The version 7 not usable for me anymore.

**Comment by lwouis** [comment]
Today I checked out https://github.com/ejbills/DockDoor. It was such an interesting experience. I went to check out some specific topic, because a user mentioned it. I ended up checking out the Settings of this app, to see how they were doing compared to AltTab.

This app went the same path that I went before v7: provide maximum settings so that users can customize every detail to the fullest. Looking back, I can't believe how much we improved AltTab thanks to the incredible work and drive from @Allsochen. We went from a mess of settings to something way more approachable.

Just look at these settings:

<img width="1020" height="894" alt="Image" src="https://github.com/user-attachments/assets/d05a6bc5-580d-4bf2-92f6-2c93981b4cb6" />

Try to actually read them one by one. See how long it takes, and the mental energy it requires to actually understand each one. Notice the lack of hierarchy and importance. Notice the size of the scrollbar. And in case it's not complex enough, there is an Advanced Settings button!

<img width="976" height="850" alt="Image" src="https://github.com/user-attachments/assets/4f725770-be45-4d0d-a52d-131fd1d0f972" />

There is still work to do to improve AltTab for sure. Looking at this DockDoor app, I'm really happy about the change in philosophy we took. I understand it frustrates the most demanding users. This is not a great feeling. However, we should celebrate how accessible it makes the important customizations, to most people. Just look at this. What a success:

<img width="672" height="716" alt="Image" src="https://github.com/user-attachments/assets/d92ce766-85d0-423e-94f4-6fab07f19000" />

As a next step, I think we would benefit from having an onboarding experience. Something that pops once, and guide the user to configure AltTab to their liking, in broad strokes.

Also, a review of hierarchy. We can probably burry some settings (Controls tab is still huge), and bring others to the top (e.g. Preview feature).

Thank you

**Comment by phazei** [comment]
What an excuse not to add a 4th shortcut option.  Options are needed to provide functionality, you look at them once, set them up, and never think about them again.  I had 4 shortcuts, and didn't look or think about it for a year.  Then it broke and what had been one of the useful programs on my entire system became useless for my workflow since I wasn't able to navigate as I needed.  Now I'm stuck on a very old version like half the other people here because someone wants things to look pretty.  They're the settings, they're under the hood.  Yeah, make the basic settings pretty and nice, but there should be settings beyond that to provide actual functionality.  If VLC didn't have the "simple vs all" settings choice, it would be useless.  If windows didn't have control panel, it's basic settings would be useless.  Burry if you must, that's fine, but don't cripple, just make the extra settings deeper.

**Comment by rosmur** [comment]
VS code does settings right by a huge margin:

- searchable
- all items in 1 pane but clearly categorized into sections allowing easy visual navigation
- UI and declarative config file option 
- given you have access to a JSON, it enables a powerful new mode of just asking AI to change a setting for you. 

Not saying AltTab needs to go this path, but just sharing this as an example of what I think the gold standard is for settings UX.

**Comment by TheophileMot** [comment]
It's certainly commendable to make a clean, organized UI, and we can appreciate the design and coding effort involved. However, it's unfortunate that the update has significantly reduced the functionality of the app; for me, at least, it defeats the purpose of providing a customizable app, and I'm still happily using v6.

I think a big question is: how often do users actually need to tinker with the settings page? I'd guess the answer is close to "once": you go to the settings, read through a small number of options (even in v6, there aren't that many, and they're very understandable), set them exactly how you like them, and never look back. From that point of view, it looks like the update is a solution in search of a problem.

If some users need to switch layouts for whatever reason without worrying about the details, then sure, provide some convenient buttons with reasonable presets. But why prevent the rest of us from making more nuanced choices?

**Comment by lwouis** [comment]
I can't believe the tone from @phazei
> What an excuse

> because someone wants things to look pretty

Is this normal to talk to strangers like this where you live? From my perspective this is very disrespectful. I would never talk like this to someone I don't know. Nor someone I know. You should show more respect to people when conversing.

I find myself in disagreement with @rosmur and @TheophileMot, on where to go with the software, however I find their messages very respectful and helpful. They make me think and consider things. They offer a different perspective. They show respect and consideration.

There is no place for such behavior in public, @phazei. Please be more respectful of others.

Also, even on a self-centered/selfish level, think for a moment. Do you really think antagonizing people, and being rude, will get you what you want? Who wants to be surrounded with people that are not nice? If you want to work with people, they first have to like you.

**Comment by phazei** [comment]
@lwouis 
I thought it was curt, but not on the realm of disrespectful, there are many adjectives that could have gone before the word 'excuse', but it was left neutral, no disrespect intended.  I was an observation based on message and prior messages stating similar reasoning.

What I find disrespectful is when some spends a couple days working on a PR and starting a discourse on it, having had to learn swift and the entire process of how to build an app for MacOS in xcode.  Then spending while writing a in depth reply discussing the minutiae of the matter to better refine it, and then getting ignored.  Or discarding the thoughts of the most passionate users of an app that want to participate and help make it better when there's been more outspoken requests for a particular thing that's relatively easy to implement and only reason for the contrary is aesthetics.

**Comment by azhongzl** [comment]
@lwouis Thanks for the great work! It helps a lot! Just one thing, I'd like each shortcut to have its own appearance settings. For example, shortcut 1 uses "App Icons", while shortcut 2 uses "Thumbnails". I use different appearance settings for switching between apps and switching between all windows in one app.

**Comment by JohnnyFireOne** [comment]
count me in for restoring the customization!

**Comment by lwouis** [comment]
Hi,

Could you please help me QA [this build](https://github.com/user-attachments/files/25328428/AltTab.app.zip)? It adds custom number of shortcuts with the great design suggested by @Elijas here: https://github.com/lwouis/alt-tab-macos/issues/3882#issuecomment-2888247094

Note that it also contains all the recent work (see https://github.com/lwouis/alt-tab-macos/issues/5255#issuecomment-3904464703)

Thank you

**Comment by azhongzl** [comment]
The new build is great! But is it possible to allow each shortcut to have its own appearance settings? I'm asking because I want to use:
- Shortcut 1 to switch between apps (not windows)
- Shortcut 2 to switch between all windows in an app (not apps)

**Comment by lwouis** [comment]
@azhongzl how would the UI look like? This is a puzzle I haven't been able to solve. If almost every setting is per-shortcut, then I can't think of a way to present the Settings UI that wouldn't be super confusing for users 😅

**Comment by azhongzl** [comment]
Yes, that's true. Maybe it's too complex for others to use (just fit in my case :D).

Could it be possible to have several appearance settings and shortcut settings, and in each shortcut, you can choose a predefined appearance? By default, the shortcut would use the first appearance setting (for users with default settings).

**Comment by vinodt1** [comment]
@lwouis Please also make the hotkey optional for each shortcut. This way we can trigger them via scripts like Hammerspoon using the CLI commands without exhausting the available hotkeys system wide. This will facilitate multiple use cases in multiple scenarios.

**Comment by lwouis** [comment]
@vinodt1 isn't it already the case? if you click `+` it will add a new shortcut with no key. you can also clear a key of an existing shortcut. this makes is available via CLI while not reserving a global shortcut

**Comment by vinodt1** [comment]
> [@vinodt1](https://github.com/vinodt1) isn't it already the case? if you click `+` it will add a new shortcut with no key. you can also clear a key of an existing shortcut. this makes is available via CLI while not reserving a global shortcut

Sorry, I didn't test the new build before. Yes, it now makes the hotkey optional. But one thing that is related to this is the appearance setting for where to show the UI. Now it is a common setting for all shortcuts. To use the app with mouse gestures in BTT, I would need to be able to change this setting to appear on the screen with the mouse. But for shortcuts that use the hotkey I would need to be able to trigger it on the screen with active window. 

I was testing the search functionality also. I have some thoughts about this:
1. after typing I'm not able to select the window using the enter key which is more intuitive in this case. I could only select with space key.
2. The window filtering is not helping since a single keyword is filtering multiple windows in my testing. It might be doing the partial matching and showing all the windows it could and the window I'm interested is not the first in the list.
3. Since we are on this topic, maybe for a future release, the faster way to select a window is with vimium style key hints. That would be way faster than typing and searching, since a single keyword can filter multiple windows.

**Comment by enderunal** [comment]
Hey @lwouis ,

Thanks for the new build. I couldn't identify any issues.

On the other hand, my main blocker still is the new big UI as I explained before. 

This is how it's look like since new version:
<img width="492" height="134" alt="Image" src="https://github.com/user-attachments/assets/1ef58883-f7ae-4b65-9e22-22fd2158dbb1" />


And this is how it looks like in the pre v7. 
<img width="914" height="64" alt="Image" src="https://github.com/user-attachments/assets/c14d3834-94ec-4ff1-9a35-44923e8f67c9" />

The main differences are the icon sizes are too big and you don't see the other titles unless you scroll to them.
I'll now switch back again to older version.

**Comment by lwouis** [comment]
Regarding the feature of having some global settings be per-shortcut instead, I've opened this dedicated ticket, to discuss UX solutions: https://github.com/lwouis/alt-tab-macos/issues/5313 👍 

I've done a lot of the feature from this thread. I think it may be nicer to close it, and open dedicated tickets.

**Comment by Elijas** [comment]
Hey @lwouis thank you so much for engaging with the community

I feel like we could satisfy a lot of the requests here with this kind of design, while keeping the app super simple to use and understsand, what do you think?

https://github.com/user-attachments/assets/c75cf9a5-4155-4566-91a3-67a7fb074129

### [→ More details in #5313](https://github.com/lwouis/alt-tab-macos/issues/5313#issuecomment-3925890507)

**Comment by Elijas** [comment]
> Hey [@lwouis](https://github.com/lwouis) thank you so much for engaging with the community
> 
> I feel like we could satisfy a lot of the requests here with this kind of design, while keeping the app super simple to use and understsand, what do you think?
> 
>  Screen.Recording.2026-02-19.at.11.02.12.mov 
> ### [→ More details in #5313](https://github.com/lwouis/alt-tab-macos/issues/5313#issuecomment-3925890507)

@billy-bg @TheophileMot @felixier @phazei @azhongzl feedback [in this thread](https://github.com/lwouis/alt-tab-macos/issues/5313#issuecomment-3925890507) would be highly appreciated, as you were previously interested in this improvement

**Comment by shawngmc** [comment]
I'd really like to see some icon scaling controls. Never used the old version; but I end up with very small icons when I have a number of windows open.

**Comment by StFS** [comment]
> @lwouis was the option to "Hide apps with no open windows" removed? I can't find it in my settings UI.
> 
> I'd relly like to get that back, I use AltTab to switch between open windows on my active desktop only. If I want to switch to an app that doesn't have any open windows I use the MacOS Cmd+Tab.
> 
> This is the option I'm talking about:
> https://github.com/lwouis/alt-tab-macos/issues/4225

EDIT: nevermind... I found it

<img width="710" height="504" alt="Image" src="https://github.com/user-attachments/assets/691780e6-bea7-40c3-bd69-5bec0f4952a0" />

**Comment by olegcherr** [comment]
Thanks for the app!
I personally use the Titles-only mode (list view).
Unfortunately, the latest versions are quite ugly 😃 (sorry)
I'd be happy if you could restore the layout settings or improve the appearance of the list.
v6 | v11
-|-
bigger and easier to read <img width="2560" height="1920" alt="Image" src="https://github.com/user-attachments/assets/6345236b-b235-4c64-81d0-d865f554883f" /> | small (even when set to Large size) and too wide. placed at exact screen center, which is actually not a perfect place. should be shifted little bit to the top for more eye comfort <img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/2f7c255c-59b2-4baf-ae72-943317ce3e76" />

**Comment by mehdi-dbx** [comment]
please restore this feature,
pleaaaaase ...

_- Adding an XS and XL Appearance Size => sounds like we should add it._

this was the main reason I switched to alt-tab
I have 50+ app windows open and this ends up eating the whole screen real estate and eye-tiring to look for the app id like to switch to instead of locating it with the blink of an eye
I miss the custom size slider sooooo much

tbh i would stick to pre 7 if I could but its not compatible with my OS anymore ...

**Comment by lwouis** [comment]
Hi @mehdi-dbx,

Could you please clarify a bit what your ideal experience would be?

You'd like smaller thumbnails, so more fit on the screen at once? You also said :

> eye-tiring to look for the app id like to switch to instead of locating it with the blink of an eye

Do you mean that's it's tiring that you have to scroll the grid currently?

Thank you

**Comment by mehdi-dbx** [comment]
Thanks for you quick reply L

_> Could you please clarify a bit what your ideal experience would be?_
_> You'd like smaller thumbnails, so more fit on the screen at once?_ 

Yes - that was my previous experience - talking about the App Icon view, not the tiled one
(I was about to reintroduce this feature myself in a fork and rebuild but I just deleted XCode due to its huge toll on disk size)

_> You also said :_
_> eye-tiring to look for the app id like to switch to instead of locating it with the blink of an eye_

Before, i was able to spot all my apps, concentrated at the center of the screen.
Now, even with "Small", I have to scan my large screen all the way from left to right to locate the app I'd like to switch to.

I am a long time user, but honestly this is a deal-breaker for me - I stuck to version 6 as long as I could until I couldn't anymore
I mean, I totally get the T-shirt size simplification, but I was expecting to find it in the advanced menu.

TYSM for the support (and the awesome app)

**Comment by lwouis** [comment]
Thank you @mehdi-dbx 

Just to be 100% sure I understand. Your preference is not necessarily to have the thumbnails be smaller, so that more fit in the current switcher. Your preference is that the switcher itself is smaller, so the area it covers it physically smaller on your screen, and then have the thumbnails smaller so they fit better on that small rectangle, and there is no scroll.

Is my understanding correct?

Currently, I've set an algorithm that estimate the user field-of-view, based on their screen size, and an estimate of how much they can see without moving their head, depending on if they are in front of a laptop, monitor, TV, etc.

Then the switcher is shown as large as possible in that field-of-view.

I'm thinking that for your use-case, and to help other people, we could perhaps introduce not a new size, but a "Compact mode", where the switcher itself takes less space on screen, and the size of its contents scale down accordingly.

Thank you

**Comment by mehdi-dbx** [comment]
@lwouis, 


> Just to be 100% sure I understand. Your preference is not necessarily to have the thumbnails be smaller, so that more fit in the current switcher. Your preference is that the switcher itself is smaller, so the area it covers it physically smaller on your screen, and then have the thumbnails smaller so they fit better on that small rectangle, and there is no scroll.
> 
> Is my understanding correct?
 

Yes that is correct

I think its a good thing to leave choice to user in advanced setting
I see the rationale behind fitting the switcher to be shown as large as possible in the field-of-view, 
but for some users with a lot of apps open having all of them visible in a condensed field at the center of the screen takes less effort for the eyes and brain (versus scanning the whole screen width with your eyes to locate that app you want to switch to)

This was a key feature for me and Im not updating past Version 6.60.0 because it's real deal breaker for me.

Thank your for your efforts maintaining this project, its very useful to tons of people
Regards
